src/components/Skill/Skill.tsx

```diff
@@ -0,0 +1,623 @@
+            !e.currentTarget.contains(e.relatedTarget)
+          ) {
+            setSkillExpanded(false);
+          }
+        }}
+        onFocus={(e: React.FocusEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onFocus?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(true);
+          }
+        }}
+        onKeyDown={(e: React.KeyboardEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onKeyDown?.(e);
+          if (
+            clickable &&
+            expandable &&
+            (e.key === eventKeys.ENTER || e.key === eventKeys.SPACE)
+          ) {
+            if (e.key === eventKeys.SPACE) {
+              e.preventDefault(); // Prevents page scroll
+            }
+            setSkillExpanded((value) => !value);
+          }
+        }}
+        onMouseEnter={(e: React.MouseEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onMouseEnter?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(true);
+          }
+        }}
+        onMouseLeave={(e: React.MouseEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onMouseLeave?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(false);
+          }
+        }}
+        style={{ ...style, color, height, width }}
+        tabIndex={tabIndex}
+        title={title}
+        ref={mergedRef}
+      >
+        <div
+          className={styles.background}
+          onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+            if (
+              readonly ||
+              disabled ||
+              !clickable ||
+              (variant !== SkillVariant.Block && size === SkillSize.XSmall) ||
+              e.target !== e.currentTarget
+            ) {
+              return;
+            }
+            onClick?.(e);
+            if (expandable && variant === SkillVariant.Block) {
+              setSkillExpanded((value) => !value);
+            }
+          }}
+          style={{ background }}
+        ></div>
+        <div className={styles.content}>{children}</div>
+        {variant === SkillVariant.Block && (
+          <>
+            {skillExpanded && <div className={styles.dividerExpandedSkill} />}
```

Comment:
In PCS Profile review, instead of just an 'expanded' content section, we have more of a read only and an edit section. When a skill card is collapsed, we can show things like related experience and text as paragraphs, which take up less space, and when they expand we can render them as select and input fields. Both of those views sit underneath the skill, slider, and menu pieces. 

How would I build something similar with this component?

Reply:
Read the expanded state via your useState implementation, update the `content` and `expandedContent` of the Skill Block.